src/status_im/contexts/wallet/wallet_connect/events.cljs

```diff
@@ -228,3 +228,33 @@
             :params     [(js/JSON.stringify session-info)]
             :on-success #(log/info "Wallet Connect session persisted")
             :on-error   #(log/info "Wallet Connect session persistence failed" %)}]]]}))
+
+(rf/reg-event-fx
+ :wallet-connect/if-need-to-handle-session-request
+ (fn [{:keys [db]} [event on-handle on-do-not-handle]]
+   (let [{:keys [topic]} event
+         should-handle?  (some #(= topic %)
+                               (map :topic (:wallet-connect/sessions db)))]
+     (if should-handle?
+       {:fx [[:dispatch on-handle]]}
+       {:fx [[:dispatch on-do-not-handle]
+             [:dispatch
+              [:wallet-connect/send-response
+               {:error (wallet-connect/get-sdk-error
+                        constants/wallet-connect-user-rejected-chains-error-key)}]]]}))))
```

Comment:
wait, but what is the usage of it? I see you added an event but I don't see you change anything to dispatch this event somewhere

Reply:
he will add it on top of your changes when they're merged from what I got. @shivekkhurana would be better for you to rebase Alex's branch onto yours and change the PR to merge into it instead of develop (so we only see your changes and it will change to develop automatically once his is merged).